src/ui/hexedit.cc

```diff
@@ -1341,24 +1346,49 @@ void HexEdit::saveChunkToFile(const QString& path) {
   saveDataToFile(byte_offset, size, path);
 }
 
+/**
+ * The data is saved to temporary file and then moved to destined location,
+ * so that the previous content wouldn't be destroyed if something went wrong
+ * (e.g. connection with server was broken and couldn't load data to save).
+ */
 void HexEdit::saveDataToFile(qint64 byte_offset, qint64 size,
                              const QString& path) {
   if (byte_offset + size > dataBytesCount_) {
     size = dataBytesCount_ - byte_offset;
   }
+  auto data = edit_engine_.bytesValues(byte_offset, size);
 
-  auto data_to_save = edit_engine_.bytesValues(byte_offset, size);
+  QString tmp_path = path + ".tmp." + util::generateRandomUppercaseText(8);
 
-  QFile file(path);
+  // It will be closed automatically.
```

Comment:
No need to say this, this is a general convention used everywhere in C++.

Reply:
Removed